pycloudlib/utils/backoff.py

```diff
@@ -0,0 +1,70 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""Backoff util to retry a function with exponential backoff for specific exceptions."""
+
+import random
+import time
+from functools import wraps
+from typing import Tuple, Type
+
+from pycloudlib.errors import PycloudlibTimeoutError
+
+
+def exponential_backoff(
```

Comment:
Nit suggestion:

```suggestion
def exponential_backoff(
    *,
```

This, forces client code to always give args as kwargs and, IMO, improves readability in call-sites.